frontend/src/app/services/auth-page-guard/auth-page-guard.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot} from '@angular/router';
+import { Observable, of } from 'rxjs';
+import { RouteName } from 'src/app/enums/route-name-enun';
+import { AuthService } from '../auth/auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthPageGuardService implements CanActivate {
```

Comment:
In guard name you don't need to use AuthPage, this guard not only for auth page. It uses for a lot of other pages